scripts/eachdist.py

```diff
@@ -0,0 +1,371 @@
+
+    if args.all:
+        allstr = args.allsep.join(
+            fmt_for_path(args.all, path) for path in targets
+        )
+        cmd = args.format.format(allstr)
+        _runcmd(cmd)
+    else:
+        for target in targets:
+            cmd = fmt_for_path(args.format, target)
+            _runcmd(cmd)
+
+
+def clean_remainder_args(remainder_args):
+    if remainder_args and remainder_args[0] == "--":
+        del remainder_args[0]
+
+
+def join_args(arglist):
+    return " ".join(map(shlex.quote, arglist))
+
+
+def install_args(args):
+    clean_remainder_args(args.pipargs)
+    if args.eager_upgrades:
+        args.pipargs += ["--upgrade-strategy=eager"]
+
+    if args.with_dev_deps:
+        runsubprocess(
+            args.dry_run,
+            [
+                "python",
+                "-m",
+                "pip",
+                "install",
+                "--upgrade",
+                "pip",
+                "setuptools",
+                "wheel",
+            ]
+            + args.pipargs,
+            check=True,
+        )
+
+    allfmt = "-e 'file://{}'" if args.editable else "'file://{}'"
+    execute_args(
+        parse_subargs(
+            args,
+            (
+                "exec",
+                "python -m pip install {} " + join_args(args.pipargs),
+                "--all",
+                allfmt,
+            ),
+        )
+    )
+    if args.with_dev_deps:
+        rootpath = find_projectroot()
+        runsubprocess(
+            args.dry_run,
+            [
+                "python",
+                "-m",
+                "pip",
+                "install",
+                "--upgrade",
+                "-r",
+                str(rootpath / "dev-requirements.txt"),
+            ]
+            + args.pipargs,
+            check=True,
+        )
+
+
+def parse_subargs(parentargs, args):
+    subargs = parse_args(args)
+    subargs.dry_run = parentargs.dry_run or subargs.dry_run
+    return subargs
+
+
+def lint_args(args):
+    rootdir = str(find_projectroot())
+
+    runsubprocess(
+        args.dry_run,
+        ("black", ".") + (("--diff", "--check") if args.check_only else ()),
+        cwd=rootdir,
+        check=True,
+    )
+    runsubprocess(
+        args.dry_run,
+        ("isort", "--recursive", ".")
+        + (("--diff", "--check-only") if args.check_only else ()),
+        cwd=rootdir,
+        check=True,
+    )
+    runsubprocess(args.dry_run, ("flake8", rootdir), check=True)
+    execute_args(
+        parse_subargs(
+            args, ("exec", "pylint {}", "--all", "--mode", "lintroots",),
+        )
+    )
+
+
+def test_args(args):
+    clean_remainder_args(args.pytestargs)
+    execute_args(
+        parse_subargs(
+            args,
+            (
+                "exec",
+                "pytest {} " + join_args(args.pytestargs),
+                "--mode",
+                "testroots",
+            ),
+        )
+    )
+
+
+def main():
+    args = parse_args()
+    args.func(args)
```

Comment:
Small change suggestion, makes it easier to find help when someone runs the script the first time.

```suggestion
    
    try:
        args.func(args)
    except AttributeError:
        print("Invalid argument. {} -h for usage".format(__file__))
        sys.exit(1)
```

Reply:
That's actually quite a bad behavior I have here, thanks for noticing!

Reply:
I fixed it based on https://stackoverflow.com/a/23354355/2128694. Now the output when calling without command is

```
$ scripts/eachdist.py
usage: eachdist.py [-h] [--dry-run] command ...
eachdist.py: error: the following arguments are required: command
```

Reply:
👍